test/distributed/org/apache/cassandra/distributed/test/SSTableGenerationTest.java

```diff
@@ -0,0 +1,177 @@
+    {
+        try (Cluster cluster = init(Cluster.build(1).withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, true)).start()))
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, v int, primary key (pk))");
+            createSSTable(cluster);
+            createSSTable(cluster);
+            assertSSTablesCount(cluster, 0, 2);
+            verfiySSTableActivity(cluster, false);
+
+            waitOn(cluster.get(1).shutdown(true));
+            cluster.get(1).config().set(ENABLE_UUID_FIELD_NAME, false);
+
+            Assertions.assertThatExceptionOfType(RuntimeException.class)
+                      .isThrownBy(() -> cluster.get(1).startup())
+                      .withCauseInstanceOf(SystemExitException.class);
+        }
+    }
+
+
+    private void verfiySSTableActivity(Cluster cluster, boolean expectLegacyTableIsPopulated)
+    {
+        cluster.get(1).runOnInstance(() -> {
+            RestorableMeter meter = new RestorableMeter(15, 120);
+            SequenceBasedSSTableUniqueIdentifier seqGenId = new SequenceBasedSSTableUniqueIdentifier(1);
+            SystemKeyspace.persistSSTableReadMeter("ks", "tab", seqGenId, meter);
+            assertThat(SystemKeyspace.getSSTableReadMeter("ks", "tab", seqGenId)).matches(m -> m.fifteenMinuteRate() == meter.fifteenMinuteRate() && m.twoHourRate() == meter.twoHourRate());
+
+            checkSSTableActivityRow(SSTABLE_ACTIVITY_V2, seqGenId.asBytes(), true);
+            if (expectLegacyTableIsPopulated)
+                checkSSTableActivityRow(LEGACY_SSTABLE_ACTIVITY, seqGenId.generation, true);
+
+            SystemKeyspace.clearSSTableReadMeter("ks", "tab", seqGenId);
+
+            checkSSTableActivityRow(SSTABLE_ACTIVITY_V2, seqGenId.asBytes(), false);
+            if (expectLegacyTableIsPopulated)
+                checkSSTableActivityRow(LEGACY_SSTABLE_ACTIVITY, seqGenId.generation, false);
+
+
+            UUIDBasedSSTableUniqueIdentifier uuidGenId = new UUIDBasedSSTableUniqueIdentifier(UUIDGen.getTimeUUID());
+            SystemKeyspace.persistSSTableReadMeter("ks", "tab", uuidGenId, meter);
+            assertThat(SystemKeyspace.getSSTableReadMeter("ks", "tab", uuidGenId)).matches(m -> m.fifteenMinuteRate() == meter.fifteenMinuteRate() && m.twoHourRate() == meter.twoHourRate());
+
+            checkSSTableActivityRow(SSTABLE_ACTIVITY_V2, uuidGenId.asBytes(), true);
+
+            SystemKeyspace.clearSSTableReadMeter("ks", "tab", uuidGenId);
+
+            checkSSTableActivityRow(SSTABLE_ACTIVITY_V2, uuidGenId.asBytes(), false);
+        });
+    }
+
+    private static void checkSSTableActivityRow(String table, Object genId, boolean expectExists)
+    {
+        String cql = "SELECT rate_15m, rate_120m FROM system.%s WHERE keyspace_name=? and columnfamily_name=? and generation=?";
+        UntypedResultSet results = executeInternal(String.format(cql, table), "ks", "tab", genId);
```

Comment:
Nit: can we assert that `results` is not null?
```suggestion
        UntypedResultSet results = executeInternal(String.format(cql, table), "ks", "tab", genId);
        Assert.assertNotNull(results);
```